pkg/host/host.go

```diff
@@ -0,0 +1,423 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package host
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+)
+
+const (
+	hostPathFromDaemon    = "/host"
+	redhatReleaseFile     = "/etc/redhat-release"
+	rhelRDMAConditionFile = "/usr/libexec/rdma-init-kernel"
+	rhelRDMAServiceName   = "rdma"
+	rhelPackageManager    = "yum"
+
+	ubuntuRDMAConditionFile = "/usr/sbin/rdma-ndd"
+	ubuntuRDMAServiceName   = "rdma-ndd"
+	ubuntuPackageManager    = "apt-get"
+
+	genericOSReleaseFile = "/etc/os-release"
+)
+
+// Contains all the host manipulation functions
+//
+//go:generate ../../bin/mockgen -destination mock/mock_host.go -source host.go
+type HostManagerInterface interface {
+	TryEnableTun()
+	TryEnableVhostNet()
+	TryEnableRdma() (bool, error)
+
+	// private functions
+	// part of the interface for the mock generation
+	LoadKernelModule(name string, args ...string) error
+	isRHELSystem() (bool, error)
+	isUbuntuSystem() (bool, error)
+	isCoreOS() (bool, error)
+	rdmaIsLoaded() (bool, error)
+	enableRDMA(string, string, string) (bool, error)
+	installRDMA(string) error
+	triggerUdevEvent() error
+	enableRDMAOnRHELMachine() (bool, error)
+	getOSPrettyName() (string, error)
+}
+
+type HostManager struct {
+	RunOnHost bool
+	cmd       utils.CommandInterface
+}
+
+func NewHostManager(runOnHost bool) HostManagerInterface {
+	return &HostManager{
+		RunOnHost: runOnHost,
+		cmd:       &utils.Command{},
+	}
+}
+
+func (h *HostManager) LoadKernelModule(name string, args ...string) error {
```

Comment:
maybe it's better to use https://github.com/pmorjan/kmod but I'm not sure on it

Reply:
that lib doesn't support chroot. but maybe in the future we can switch this implementation with something like that (add a support for that lib to use chroot methods)